src/main/scala/org/apache/spark/ml/feature/RapidsPCA.scala

```diff
@@ -168,26 +185,56 @@ class RapidsPCAModel(
    */
   override def transform(dataset: Dataset[_]): DataFrame = {
     val outputSchema = transformSchema(dataset.schema, logging = true)
+    val gpuIdBC = dataset.rdd.context.broadcast(getGpuId)
 
-    // TODO(rongou): make this faster and re-enable.
-    //    if (getUseGemm) {
-    //      val transformed = dataset.toDF().rdd.mapPartitions(iterator => {
-    //        val gpuID = TaskContext.get().resources()("gpu").addresses(0)
-    //        val partition = iterator.toList
-    //        val A = Matrices.fromVectors(partition.map(_.getAs[Vector]($(inputCol)))).toDense
-    //        val C = Matrices.zeros(partition.length, getK).toDense
-    //        CUBLAS.gemm_b(A, pc, C, gpuID)
-    //        C.rowIter.zip(partition.iterator).map { case (v, r) =>
-    //          Row.fromSeq(r.toSeq ++ Seq(v))
-    //        }
-    //      })
-    //      dataset.sparkSession.createDataFrame(transformed, outputSchema)
-    //    }
-    //    else {
-    val transposed = pc.transpose
-    val transformer = udf { vector: Vector => transposed.multiply(vector) }
-    dataset.withColumn($(outputCol), transformer(col($(inputCol))), outputSchema($(outputCol)).metadata)
-    //    }
+    /**
+     * UDF class to speedup transform process of PCA
+     */
+    class gpuTransform extends Function[mutable.WrappedArray[Double], Array[Double]]
+      with RapidsUDF with Serializable {
+      override def evaluateColumnar(args: ColumnVector*): ColumnVector = {
+        logDebug("==========using GPU transform==========")
+        val gpu = if (gpuIdBC.value == -1) {
+          TaskContext.get().resources()("gpu").addresses(0).toInt
+        } else {
+          gpuIdBC.value
+        }
+        require(args.length == 1, s"Unexpected argument count: ${args.length}")
+        val input = args.head
+        val input_rows = input.getRowCount.toInt
+        // Due to the layout of LIST type ColumnVector, cublas gemm function should return the transposed result matrix
+        // for compatibility. e.g. an expected output matrix(actually a columnar vector of LIST type)
+        // [1,2]
+        // [3,4]
+        // [5,6]
+        // its memory data layout from Cublas GEMM is [1,3,5,2,4,6]. However, it will be displayed in LIST ColumnVector as :
+        // [1,3]
+        // [5,2]
+        // [4,6]
+        // To fill the gap between native memory and CV data storage, we consider the following nature:
+        // when A * B = C, we get BT * AT = CT. (T means transpose). In this case the output matrix becomes:
+        // [1,3,5]
+        // [2,4,6]
+        // whose memory data layout is [1,2,3,4,5,6]. Then it can be comsumed by CV directly.
+        val C = RAPIDSML.gemmWithColumnViewPointer(pc, input, input_rows ,gpu)
```

Comment:
I like how much simpler this call now looks!  Is it necessary to pass `input_rows` here since it can be trivially derived from `input`?  Also the name is a bit out of date with the recent refactoring, should be updated to something like `gemmWithColumnView` or maybe just `gemm`.